accounts/forms.py

```diff
@@ -0,0 +1,62 @@
+import re
+
+from django.contrib.auth.forms import UserCreationForm, UserChangeForm
+from django import forms
+from django.contrib.auth import get_user_model
+
+
+def split_full_name(full_name):
+    if not full_name:
+        first_name, last_name = None, None
+    else:
+        parts = full_name.split(' ')
+        if len(parts) == 1:
+            first_name = parts[0]
+            last_name = None
+        elif len(parts) == 2:
+            first_name = parts[0]
+            last_name = parts[1]
+        else:
+            first_name = parts[0]
+            last_name = ' '.join(parts[1:])
+    return first_name, last_name
+
+
+class CustomUserCreationForm(UserCreationForm):
+    phone_number = forms.CharField(required=True)
+    # phone_number = forms.RegexField(regex='^((\+92)|(0092))-{0,1}\d{3}-{0,1}\d{7}$|^\d{11}$|^\d{4}-\d{7}$')
+    full_name = forms.CharField(required=False)
+
+    class Meta(UserCreationForm.Meta):
+        model = get_user_model()
+        fields = UserCreationForm.Meta.fields + ('phone_number', 'email', 'full_name', 'profile_pic_url', 'country',)
+
+    def clean(self):
+        phone_number = self.cleaned_data.get('phone_number')
+        if re.match(r"^((\+92)|(0092))-{0,1}\d{3}-{0,1}\d{7}$|^\d{11}$|^\d{4}-\d{7}$", phone_number):
+            pass
+        else:
+            raise forms.ValidationError(
+                "phone number must start with '0' or '0092' or '+92' " +
+                "and the remaining digits cannot be more than 9 digits"
+            )
+
+        return self.cleaned_data
+
+    def clean_full_name(self):
```

Comment:
clean_full_names is just to validate full name so in-here you will only do validations  like maybe you can check that field is not empty or words are separated by spaces.

To split name and save data in relevant fields of db, try to override "**save" method**.

Reply:
sure

Reply:
done